### PR TITLE
Checks new ENV vars inside worker

### PR DESCRIPTION
**Pull request by alex-litvak**

### CODE REVIEW

File: iron_worker.py [hunk 1]
```diff
@@ -544,3 +544,36 @@ def encode_multipart_formdata(fields, files):
...
+                args['payload_file'] = sys.argv[i + 1]
+            if sys.argv[i] == "-config":
+                args['config_file'] = sys.argv[i + 1]
+        
+        if os.getenv('TASK_ID') != None: args['task_id'] = os.getenv('TASK_ID')
+        if os.getenv('TASK_DIR') != None: args['dir'] = os.getenv('TASK_DIR')
+        if os.getenv('PAYLOAD_FILE') != None: args['payload_file'] = os.getenv('PAYLOAD_FILE')
+        if os.getenv('CONFIG_FILE') != None: args['config_file'] = os.getenv('CONFIG_FILE')
+
+        if args.has_key('payload_file') and file_exists(args['payload_file']):
```
Comment (ulandj) [inline iron_worker.py]: @alex-litvak use `if 'payload_file' in args` instead of `has_key`. `has_key` is deprecated.

File: iron_worker.py [hunk 2]
```diff
@@ -544,3 +544,36 @@ def encode_multipart_formdata(fields, files):
     @staticmethod
     def get_content_type(filename):
         return mimetypes.guess_type(filename)[0] or 'application/octet-stream'
+
+    def get_args(self):
```
Comment (ulandj) [inline iron_worker.py]: where this method will be called?

Comment (alex-litvak) [inline iron_worker.py]: For example, his method can be called inside worker to receive config, payload, etc.

File: iron_worker.py [hunk 3]
```diff
@@ -544,3 +544,36 @@ def encode_multipart_formdata(fields, files):
...
+        global args
+        args = {'task_id': None, 'dir': None, 'payload': [], 'config': None}
```
Comment (ulandj) [inline iron_worker.py]: are you sure that the `payload` must be a list (an array)? Maybe just `None`?

File: iron_worker.py [hunk 4]
```diff
@@ -544,3 +544,36 @@ def encode_multipart_formdata(fields, files):
...
+            if sys.argv[i] == "-id":
+                args['task_id'] = sys.argv[i + 1]
+            if sys.argv[i] == "-d":
+                args['dir'] = sys.argv[i + 1]
+            if sys.argv[i] == "-payload":
+                args['payload_file'] = sys.argv[i + 1]
+            if sys.argv[i] == "-config":
+                args['config_file'] = sys.argv[i + 1]
+        
+        if os.getenv('TASK_ID') != None: args['task_id'] = os.getenv('TASK_ID')
```
Comment (ulandj) [inline iron_worker.py]: The better way: 
```python
if os.getenv('TASK_ID'): 
    args['task_id'] = os.getenv('TASK_ID')
```

File: iron_worker.py [hunk 5]
```diff
@@ -544,3 +544,36 @@ def encode_multipart_formdata(fields, files):
...
+            f = open(args['payload_file'], "r")
+            contents = f.read()
+            f.close()
+            args['payload'] = json.loads(contents)
+
+        if args.has_key('config_file') and file_exists(args['config_file']):
+            f = open(args['config_file'], "r")
+            contents = f.read()
+            f.close()
+            args['config'] = json.loads(contents)
```
Comment (ulandj) [inline iron_worker.py]: The better way:
```python
    with open(args['config_file']) as content:
        args['config'] = json.load(content)
```

File: iron_worker.py [hunk 6]
```diff
@@ -238,7 +238,7 @@ def zip(self, destination=None, overwrite=True):
 
 class IronWorker:
     NAME = "iron_worker_python"
-    VERSION = "1.2.0"
+    VERSION = "1.3.0"
```
Comment (ulandj) [inline iron_worker.py]: @alex-litvak `loaded` property here

File: iron_worker.py [hunk 7]
```diff
@@ -248,6 +248,44 @@ def __init__(self, **kwargs):
...
+
+        if 'config_file' in args and file_exists(args['config_file']):
+            f = open(args['config_file'])
+            try:
+                content = f.read()
+                f.close()
+                args['config'] = json.loads(content)
+            except Exception, e:
+                print "Couldn't parse IronWorker config into json. %s" % e
+
```
Comment (ulandj) [inline iron_worker.py]: set loaded property to True here `self.loaded = True`

File: iron_worker.py [hunk 8]
```diff
@@ -507,6 +545,25 @@ def cancel(self, id, scheduled=False):
         resp = self.client.post(url)
         return True
 
+    def payload(self):
+        args = self.__get_args()
```
Comment (ulandj) [inline iron_worker.py]: without `loaded` property `__get_args()` will be read `payload` and `config` files each time when it's calling

File: iron_worker.py [hunk 9]
```diff
@@ -248,6 +248,44 @@ def __init__(self, **kwargs):
         self.client = iron_core.IronClient(name=IronWorker.NAME,
                 version=IronWorker.VERSION, product="iron_worker", **kwargs)
 
+    def __get_args(self):
```
Comment (ulandj) [inline iron_worker.py]: here `if self.loaded: return`

**Comment by ulandj** [comment]
@alex-litvak Thanks! Looks good for me. @treeder could you also take a look?

**Pull request merged**